server/test/unit/controllers/mailing.spec.js

```diff
@@ -166,5 +166,123 @@ describe('mailing controllers', () => {
         });
       });
     });
+
+    describe('sendConfirmAttendanceEmail', () => {
+      function generateProjects(amount) {
+        const projects = [];
+        for (let i = 0; i < amount; i += 1) {
+          projects.push({
+            owner: {
+              email: `owner${i}@example.com`,
+            },
+            name: `Sample Project ${i}`,
+            id: `${i}`,
+          });
+        }
+        return projects;
+      }
+
+      function generateEmailPersonalizations(amount, eventSlug, offset = 0) {
+        const personalizations = [];
+        for (let i = offset; i < amount + offset; i += 1) {
+          personalizations.push({
+            to: `owner${i}@example.com`,
+            substitutions: {
+              projectName: `Sample Project ${i}`,
+              attendingUrl: `${process.env.HOSTNAME}/events/${eventSlug}/projects/${i}/confirm-attendance?attending=true`,
+              notAttendingUrl: `${process.env.HOSTNAME}/events/${eventSlug}/projects/${i}/confirm-attendance?attending=false`,
+            },
+          });
+        }
+        return personalizations;
+      }
+
+      it('should split emails into batches of 1000 and call the mailer instance with the correct payload', () => {
```

Comment:
A last expectation to verify off by one offset would be nice (0-999), (1000-1999), (2000-2399)

Reply:
Test the indexes of the first and last of every projects to ensure that the batching is properly working